### ISSUE TITLE
Nonlinear output components

### ISSUE THREAD

**Issue by geez0x1** [geez0x1/CompliantJointToolbox#65]
Currently the nonlinear models can only apply torques internally, i.e. `x_dot = A * x + f(x)`. However in the more general sense, we could enable the more general:
```
x_dot = A * x + B * u + f(x,u);
y     = C * x + D * u + g(x,u)
```
This is necessary to implement some functionality particularly ones that affect output torque.

**Comment by geez0x1** [comment]
First working version for this can be found in `dev_nonlinear_component_outputs`.

**Comment by geez0x1** [comment]
After some more testing this has now been merged into master in 131eb8c387ec60fe044eb6ae8aa434db2f356b4f.
